File: TypeScript/codewars/Consecutive strings.ts
export function longestConsec(strarr: string[], k: number): string {
	if (strarr.length === 0 || k > strarr.length || k <= 0) {
		return "";
	}

	let longest = "";

	for (let i = 0; i <= strarr.length - k; i++) {
		const concatStr = strarr.slice(i, i + k).join("");

		if (concatStr.length > longest.length) {
			longest = concatStr;
		}
	}

	return longest;
}

File: Csharp/CodeWars/src/Completed/ConsecutiveStrings.cs
namespace CodeWars.Completed;

public class ConsecutiveStrings
{
    public static string LongestConsec(string[] strarr, int k)
    {
        if (strarr.Length == 0 || k > strarr.Length || k <= 0)
            return "";

        string longest = "";

        for (int i = 0; i <= strarr.Length - k; i++)
        {
            string concatStr = string.Concat(strarr.Skip(i).Take(k));

            if (concatStr.Length > longest.Length)
                longest = concatStr;
        }

        return longest;
    }
}
